lib/spack/spack/architecture.py

```diff
@@ -487,6 +487,27 @@ def arch_from_dict(d):
     return arch
 
 
+def get_full_system_from_platform():
```

Comment:
It looks like the tarballs use this for the name, but Spack now has `spack.platform`, and `spack.architecture.sys_type()` will give you a platform-os-target string like `darwin-elcapitan-x86_64`.  On cray the target can be more specific (`haswell`, `interlagos`, etc.) and we're planning to allow that level of specificity on other platforms (Linux, Mac) too.

Is that sufficient or do you guys need a separate platform descriptor for the binaries?